tests/unit/amo/components/TestAddon.js

```diff
@@ -745,4 +746,32 @@ describe('mapStateToProps', () => {
 
     expect(addon).toEqual(undefined);
   });
+
+  it('displays a badge when the addon is featured', () => {
+    const addon = { ...fakeAddon, is_featured: true };
+    const rootNode = renderAsDOMNode({ addon });
+
+    expect(rootNode.querySelector('.Badge-featured').textContent).toContain('Featured Extension');
```

Comment:
You may have seen examples of testing like this elsewhere but they are all relics of the past from before we had shallow rendering via enzyme. I would like to see more of a "unit testing" approach here where you are only making assertions about `<Badge/>`'s public properties. This is a safe approach because `Badge` already has its own unit tests. This is a robust approach because it allows the implementation of `Badge` to evolve independently. Here is a way to achieve the same result:

```js
const root = shallowRender({ addon });

expect(root.find(Badge)).toHaveProp('type', 'featured');
expect(root.find(Badge)).toHaveProp('label', 'Featured Extension');
```